From c702c45be88b5a6f8fd9db423fa8d724b27adaea Mon Sep 17 00:00:00 2001
From: Dragos Bogdan <dragos.bogdan@analog.com>
Date: Thu, 14 Jun 2018 10:20:05 +0300
Subject: [PATCH 14/25] dma: axi-dmac: Add support for interleaved cyclic
 transfers

An example when this mode can be enabled is where the controller
is used as a Video DMA.

Signed-off-by: Dragos Bogdan <dragos.bogdan@analog.com>
(cherry picked from commit 78f1a03268d1f739e5f94edb751ded19ff7d4d32)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/dma/dma-axi-dmac.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/dma/dma-axi-dmac.c b/drivers/dma/dma-axi-dmac.c
index 7729f5009b5a..9328407cf204 100644
--- a/drivers/dma/dma-axi-dmac.c
+++ b/drivers/dma/dma-axi-dmac.c
@@ -613,6 +613,9 @@ static struct dma_async_tx_descriptor *axi_dmac_prep_interleaved(
 		desc->sg[0].y_len = 1;
 	}
 
+	if (flags & DMA_CYCLIC)
+		desc->cyclic = true;
+
 	return vchan_tx_prep(&chan->vchan, &desc->vdesc, flags);
 }
 
-- 
2.17.0

